messaging/src/main/java/org/axonframework/serialization/avro/AvroConverter.java

```diff
@@ -0,0 +1,241 @@
+     * @param configurationOverride configuration customizer.
+     */
+    public AvroConverter(
+            @Nonnull SchemaStore schemaStore,
+            @Nonnull UnaryOperator<AvroConverterConfiguration> configurationOverride
+    ) {
+        var config = Objects.requireNonNull(configurationOverride, "the configurationOverride may not be null.")
+                            .apply(
+                                    new AvroConverterConfiguration(schemaStore)
+                            );
+        // include default strategy
+        if (config.includeDefaultAvroSerializationStrategies()) {
+            AvroConverterStrategy defaultStrategy = new SpecificRecordBaseConverterStrategy(
+                    config.schemaStore(),
+                    config.schemaIncompatibilityChecker()
+            );
+            config = config.converterStrategy(defaultStrategy);
+        }
+
+        // apply configuration to the strategies
+        AvroConverterStrategyConfiguration strategyConfiguration = config.avroConverterStrategyConfiguration();
+        config.strategies().forEach(strategy -> strategy.applyStrategyConfiguration(strategyConfiguration));
+
+        this.converterStrategies.addAll(config.strategies());
+        if (this.converterStrategies.isEmpty()) {
+            throw new IllegalStateException("At least one converter strategy is required, but none were configured.");
+        }
+        this.converter = new ChainingContentTypeConverter();
+        this.converter.registerConverter(new ByteArrayToGenericRecordConverter(schemaStore));
+    }
+
+    @Override
+    public boolean canConvert(@Nonnull Type sourceType,
+                              @Nonnull Type targetType) {
+        if (logger.isTraceEnabled()) {
+            logger.trace("Validating if we can convert from source type [{}] to target type [{}].",
+                         sourceType, targetType);
+        }
+        // no conversion needed
+        return sourceType.equals(targetType)
+                // e.g. bytes to generic record for upcaster
+                || converter.canConvert(sourceType, targetType)
+                // any type supported by the strategy to bytes and then by the chain -> serialization
+                || (strategyForType(sourceType) && converter.canConvert(byte[].class, targetType))
+                // chain to bytes and then to any type supported by a strategy -> deserialization
+                || (converter.canConvert(sourceType, byte[].class) && strategyForType(targetType))
+                // generic record to any type -> deserialization after upcaster
+                || (converter.canConvert(sourceType, GenericRecord.class) && strategyForType(targetType));
+    }
+
+    private boolean strategyForType(Type type) {
+        if (type instanceof Class) {
+            return converterStrategies.stream().anyMatch(it -> it.test((Class<?>) type));
+        } else {
+            return false; // currently only support types which are classes
```

Comment:
"currently" make me wonder if there is a way to make it support `Type`. Did you leave this for future investigation, perhaps? If so, we could draft an issue, with prio4 (as I don't think it is important for an obvious message-converter, which will have `Classes` anyhow).